src/hl/attribute.rs

```diff
@@ -0,0 +1,297 @@
+use std::fmt::{self, Debug};
+use std::marker::PhantomData;
+use std::ops::Deref;
+
+use hdf5_sys::{
+    h5::{H5_index_t, H5_iter_order_t},
+    h5a::{H5A_info_t, H5A_operator2_t, H5Acreate2, H5Aiterate2},
+};
+
+use crate::internal_prelude::*;
+
+/// Represents the HDF5 attribute object.
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct Attribute(Handle);
+
+impl ObjectClass for Attribute {
+    const NAME: &'static str = "attribute";
+    const VALID_TYPES: &'static [H5I_type_t] = &[H5I_ATTR];
+
+    fn from_handle(handle: Handle) -> Self {
+        Self(handle)
+    }
+
+    fn handle(&self) -> &Handle {
+        &self.0
+    }
+
+    // TODO: short_repr()
+}
+
+impl Debug for Attribute {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.debug_fmt(f)
+    }
+}
+
+impl Deref for Attribute {
+    type Target = Container;
```

Comment:
This type alias doesn't seem to be used.

Reply:
This is needed to be able to call methods like `.shape()` on an Attribute, which is quite convenient.